src/Peering/Peering/PeeringService/Prefix/DeleteAzurePeeringServicePrefixCommand.cs

```diff
@@ -0,0 +1,165 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+namespace Microsoft.Azure.PowerShell.Cmdlets.Peering.Peering
+{
+    using System;
+    using System.Management.Automation;
+
+    using Microsoft.Azure.Commands.Peering.Properties;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.Management.Peering;
+    using Microsoft.Azure.Management.Peering.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Common;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Models;
+
+    /// <summary>
+    /// New Azure InputObject Command-let
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzPeeringServicePrefix", DefaultParameterSetName = Constants.ParameterSetNameByName, SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    public class DeleteAzurePeeringServicePrefixCommand : PeeringBaseCmdlet
+    {
+        /// <summary>
+        /// Gets or sets the legacy InputObject.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            ParameterSetName = Constants.ParameterSetNameDefault,
+            HelpMessage = Constants.PrefixInputObjectHelp + "Prefix")]
+        [ValidateNotNullOrEmpty]
+        public PSPeeringServicePrefix InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets The Resource Group Name
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceGroupNameHelp,
+            ParameterSetName = Constants.ParameterSetNameByName)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets The InputObject NameMD5AuthenticationKeyHelp
+        /// </summary>
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.ParameterSetNameByName)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets The InputObject NameMD5AuthenticationKeyHelp
+        /// </summary>
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.ParameterSetNameByName)]
+        [ValidateNotNullOrEmpty]
+        public string PeeringServiceName { get; set; }
+
+        /// <summary>
+        /// The resource  id
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceIdHelp,
+            ParameterSetName = Constants.ParameterSetNameByResourceId)]
```

Comment:
@rileymckenna make sure this parameter has `ValueFromPipelineByPropertyName = true`